src/rod/builder/primitives.py

```diff
@@ -77,9 +77,9 @@ def __post_init__(self) -> None:
         """
 
         if isinstance(self.mesh_path, str):
-            extension = pathlib.PurePath(self.mesh_path).suffix
+            extension = self.mesh_path.split(".")[-1]
         elif isinstance(self.mesh_path, pathlib.Path):
-            extension = self.mesh_path.suffix
+            extension = str(self.mesh_path).split(".")[-1]
```

Comment:
Last curiosity, any specific reason to go back to string?

Reply:
It was just not working without a suffix keyword for NamedTempFile. I can easily revert back.